client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -402,24 +408,40 @@ function mediaButton( editor ) {
 		}
 
 		// Attempt to find media in Flux store
-		let media = assign( {}, MediaStore.get( site.ID, parsed.media.ID ) );
+		let media = assign( {}, MediaStore.get( selectedSite.ID, parsed.media.ID ) );
 		delete media.caption;
 		media = assign( {}, parsed.media, media );
 
+		// Compute a ratio used to order media sizes
+		const computeRatio = ( { width = media.width, height = media.height } ) => {
+			return Math.min(
+				( width / media.width ) || Infinity,
+				( height / media.height ) || Infinity
+			);
+		};
+
 		// Determine the next usable size
-		let sizeIndex = SIZE_ORDER.indexOf( parsed.appearance.size );
-		let size;
-		do {
-			sizeIndex += increment;
-			size = SIZE_ORDER[ Math.max( sizeIndex, 0 ) ];
-			const dimensions = MediaUtils.getThumbnailSizeDimensions( size, site );
-			if ( media.width > dimensions.width || media.height > dimensions.width ) {
-				break;
+		const sizeRatios = SIZE_ORDER
```

Comment:
I think it might help for future maintenance to include some inline comments explaining what's being done here.